### PR TITLE
Added optional environment variable configuration for contract repo branch

### PR DESCRIPTION
**Pull request by mehalter**
Add optional ability to change which branch the contracts are pulled for the API.

### CODE REVIEW

File: test/09_contracts.ts [hunk 1]
```diff
@@ -49,8 +57,14 @@ describe('contracts', () => {
...
 
-		mockRepo('balena-io', 'contracts', 'base-contracts', true);
-		await synchronizeContracts([{ owner: 'balena-io', name: 'contracts' }]);
+		mockRepo(
+			contractRepository.owner,
+			contractRepository.name,
+			contractRepository.branch,
+			'base-contracts',
+			true,
+		);
```
Comment (thgreasi) [inline test/09_contracts.ts:66]: NIT: Maybe it now makes sense to convert the arguments of this into an object, to make obvious what each arg is when invoking it and not have to manually spread all properties of the RepositoryInfo.
Eg:
```suggestion
		mockRepo({
			...contractRepository,
			filename: 'base-contracts',
			persist: true,
		});
```
Definitely doesn't have to be in this PR though.

File: src/features/contracts/contracts-directory.ts [hunk 2]
```diff
@@ -81,7 +81,9 @@ const normalizeAssets = async (
 };
 
 const getArchiveLinkForRepo = (repo: RepositoryInfo) => {
-	return `https://api.github.com/repos/${repo.owner}/${repo.name}/tarball`;
+	return `https://api.github.com/repos/${repo.owner}/${repo.name}/tarball/${
+		repo.branch || ''
```
Comment (thgreasi) [inline src/features/contracts/contracts-directory.ts:85]: Most (if not all) `|| ''` could probably be `?? ''` but not a blocker.

**Review by thgreasi — COMMENTED** [review]
LTGM, feel free to self-certify, even though I would prefer someone to check the toml & the tmpl first.

File: test/09_contracts.ts (hunk 1 above)
Comment (mehalter) [reply test/09_contracts.ts]: Improved this method's parameters in latest push.

File: src/features/contracts/contracts-directory.ts (hunk 2 above)
Comment (mehalter) [reply src/features/contracts/contracts-directory.ts]: Ah nice catch. Fixed this in the latest push

**Review by thgreasi — APPROVED** [review]

**Pull request merged**